Support for advertising identifier and adId consent collection

timkimadobe opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Update for ad ID features, related test cases, and update to Swift test app to use new ad ID features with real-world IDFA implementation example.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

timkimadobe committed:
WIP merge with current dev branch

timkimadobe committed:
update enum usage to reflect updated naming scheme

timkimadobe committed:
migrate tests from old feature branch, add new files to test targets

timkimadobe committed:
Update TestableExtensionRuntime to include new method required by ExtensionRuntime protocol

timkimadobe committed:
Migrate remaining test cases from original adID feature branch, and update variables/classes to reflect naming convention changes

timkimadobe committed:
Migrate IdentityTest cases

timkimadobe committed:
Merge pull request #1 from adobe/feature-merge-advertising-id

Feature merge advertising

timkimadobe committed:
Remove commented code and extra spaces

timkimadobe committed:
Update build script for SwiftLint to use PODS_ROOT path to check for executable
Update IdentityConstants with proper values
Update comments/documentation
Update IdentityStateTests assertion data dictionary access to use IdentityConstants enum values
Update copyright year to 2022 in new files
Remove dev comments
SwiftLint whitespace and indentation correction

timkimadobe committed:
Update test cases to handle existing AdID and no AdID with new AdID event and generic Identity event

timkimadobe committed:
Add isAdIdEvent to Event+Identity extension and build in check to adId retrieval
Add isAdIdEvent check to handleRequestContent
Simplify reset identities test case, remove persistedAdID all-zeros test case as there is no migration path to Edge Identity from this state

timkimadobe committed:
Update Event+Identity extension with rationale for String type check in isAdIdEvent check
Update documentation for adId

timkimadobe committed:
Update test case to improve readability and preserve event order verification
Update adId logic to only provide value sanitization

timkimadobe committed:
Remove outdated test for nil adID using direct state.updateAdvertisingIdentifier

timkimadobe committed:
Removing public access status for Event+Identity extension property

timkimadobe committed:
Update naming conventions for test cases
Implement functional tests for e2e testing

timkimadobe committed:
Simplify logical flow for detecting ad ID change
Rename test file
Add consent event type and source check, and format verification for testGenericIdentityRequest_whenValidAdId_thenEmptyAdId

timkimadobe committed:
Remove outdated test case with all-zeros start
Remove logging comments from test file

timkimadobe committed:
Move consent event dispatch to before resetting ECID; should send consent with old ECID

timkimadobe committed:
Update adId type in Event extension to non-optional String
Moved back shouldDispatchConsent check to original position, pending upstream service discussion

timkimadobe committed:
Remove nil coalescing operator for adId after its update to non-optional String

timkimadobe committed:
Remove consent event dispatch from resetIdentifiers flow; the flows should be separate from each other

timkimadobe committed:
Remove consent event check from resetIdentity test; consent is not longer handled as part of the reset flow

timkimadobe committed:
Update test comments and remove commented case

timkimadobe committed:
Migrating adID feature branch to current dev branch (#52)

* WIP merge with current dev branch

* update enum usage to reflect updated naming scheme

* migrate tests from old feature branch, add new files to test targets

* Update TestableExtensionRuntime to include new method required by ExtensionRuntime protocol

* Migrate remaining test cases from original adID feature branch, and update variables/classes to reflect naming convention changes

* Migrate IdentityTest cases

* Remove commented code and extra spaces

* Update build script for SwiftLint to use PODS_ROOT path to check for executable
Update IdentityConstants with proper values
Update comments/documentation
Update IdentityStateTests assertion data dictionary access to use IdentityConstants enum values
Update copyright year to 2022 in new files
Remove dev comments
SwiftLint whitespace and indentation correction

* Update test cases to handle existing AdID and no AdID with new AdID event and generic Identity event

* Add isAdIdEvent to Event+Identity extension and build in check to adId retrieval
Add isAdIdEvent check to handleRequestContent
Simplify reset identities test case, remove persistedAdID all-zeros test case as there is no migration path to Edge Identity from this state

* Update Event+Identity extension with rationale for String type check in isAdIdEvent check
Update documentation for adId

* Update test case to improve readability and preserve event order verification
Update adId logic to only provide value sanitization

* Remove outdated test for nil adID using direct state.updateAdvertisingIdentifier

* Removing public access status for Event+Identity extension property

* Update naming conventions for test cases
Implement functional tests for e2e testing

* Simplify logical flow for detecting ad ID change
Rename test file
Add consent event type and source check, and format verification for testGenericIdentityRequest_whenValidAdId_thenEmptyAdId

* Remove outdated test case with all-zeros start
Remove logging comments from test file

* Move consent event dispatch to before resetting ECID; should send consent with old ECID

* Update adId type in Event extension to non-optional String
Moved back shouldDispatchConsent check to original position, pending upstream service discussion

* Remove nil coalescing operator for adId after its update to non-optional String

* Remove consent event dispatch from resetIdentifiers flow; the flows should be separate from each other

* Remove consent event check from resetIdentity test; consent is not longer handled as part of the reset flow

* Update test comments and remove commented case

timkimadobe committed:
Merge branch 'adobe:feature/adIdConsent' into feature/adIdConsent

timkimadobe committed:
Update IdentityAdIDTests with helper functions to reduce boilerplate
Update comments to better reflect test case goals
Update test cases to use variables to store test strings
Update naming convention of ad ID value from old -> initial

timkimadobe committed:
Update test app to use real IDFA flow to update consent and ad ID
Update required pods for test app target adding: Edge & Edge Consent
Add NSUserTrackingUsageDescription to test app plist

timkimadobe committed:
Removing launch file ID

timkimadobe committed:
Update consent to be for ad ID consent instead of "collect"
Add get current consent button
Update comment and logs to be more accurate

timkimadobe committed:
Add AppTrackingTransparency.framework to build phases for Swift TestApp target

timkimadobe committed:
Remove manual consent update flow as it is handled by the ad ID event listener automatically
Update comments to help user set ad ID properly in the simulator case

timkimadobe committed:
Update to align with swiftlint

timkimadobe committed:
Update test helper access level and logic
Remove unused code in test app

timkimadobe committed:
Add test cases to cover ad ID event extension

timkimadobe committed:
Add test coverage cases

timkimadobe committed:
Merge pull request #54 from timkimadobe/feature/adIdConsent

Update Test App to use real IDFA flow for ad ID changes

timkimadobe committed:
Add log for modifying identifiers in protected ad ID namespaces (IDFA/GAID)

timkimadobe committed:
Merge branch 'adobe:feature/adIdConsent' into feature/adIdConsent

timkimadobe committed:
Update log for ad ID API violation to warning level; update string to be one-liner

timkimadobe committed:
Update quote type to be regular single quote

timkimadobe committed:
Update log message string construction so final log is on single line

timkimadobe committed:
Update log string to be more concise

timkimadobe committed:
Merge pull request #56 from timkimadobe/feature/adIdConsent

Adding log for trying to update identifiers for ad ID reserved namespaces

emdobrin committed:
Merge branch 'dev' into feature/adIdConsent

emdobrin committed:
Resolve merge conflicts

emdobrin reviewed (approved):
Pending merge conflicts resolved in https://github.com/adobe/aepsdk-edgeidentity-ios/pull/66

emdobrin committed:
Merge pull request #66 from emdobrin/feature/adIdConsent

Merge latest dev changes -> feature/adidConsent

emdobrin committed:
Updated test dependencies (#67)

* Updated dependencies, using staging changes for Edge

* Update testableExtensionRuntime for latest core

* Renaming for environment field

* CircleCI - update to Xcode 12

* Consent 1.0.0

timkimadobe committed:
Update ad ID view to use ad ID utils and better align with sample app implementation

timkimadobe committed:
Update requestTrackingAuthorization function to be pure request tracking flow; other functionality is handled by other methods in the util class

timkimadobe committed:
Merge branch 'feature/adIdConsent' into feature/ad-id-consent

# Conflicts:
#	Podfile.lock
#	SampleApps/TestApp/ContentView.swift

timkimadobe committed:
Remove get ad ID in request flow; use get ad ID flow instead to fetch or print value

timkimadobe committed:
Update documentation wording to defer to apple docs

timkimadobe committed:
Update documentation to reflect iOS version verbiage

timkimadobe committed:
Updating podfile.lock to use staging

timkimadobe committed:
Update logical flow for setting ad ID to check for tracking authorization and set the value accordingly

timkimadobe committed:
Update ad ID consent to align with sample app implementation (#69)

* Update ad ID view to use ad ID utils and better align with sample app implementation

* Update requestTrackingAuthorization function to be pure request tracking flow; other functionality is handled by other methods in the util class

* Remove get ad ID in request flow; use get ad ID flow instead to fetch or print value

* Update documentation wording to defer to apple docs

* Update documentation to reflect iOS version verbiage

* Updating podfile.lock to use staging

* Update logical flow for setting ad ID to check for tracking authorization and set the value accordingly

timkimadobe committed:
Update Swift test app to use app + scene delegate initialization for iOS 13 compatibility

timkimadobe committed:
Merge branch 'feature/adIdConsent' into feature/ad-id-consent

# Conflicts:
#	AEPEdgeIdentity.xcodeproj/project.pbxproj

timkimadobe committed:
Update test app to include storyboard which is required to use full screen

timkimadobe committed:
Remove TestApp App SwiftUI initialization file (replaced with app + scene delegates)
Apply swiftlint autocorrect and update Button closure syntax to avoid trailing closure violation

timkimadobe committed:
Update UIWindowScene cast to guard statement

timkimadobe committed:
Update modifier positions for consistency, remove unnecessary whitespace

timkimadobe committed:
Merge pull request #70 from timkimadobe/feature/ad-id-consent

Update Swift Test App for iOS 13 compatibility

emdobrin committed:
Select test scheme from xcodeproj (#71)

* Select test scheme from xcodeproj

* Select project scheme from xcworkspace

* Update test command to use conditional scheme name based on presence of project name conflict or not

* Update name conflict case paren string to use EXTENSION_NAME instead of PROJECT_NAME

* Update scheme variables to appropriate sources

Co-authored-by: Tim Kim <95260439+timkimadobe@users.noreply.github.com>

cacheung commented on SampleApps/TestApp/AppDelegate.swift:
```diff
@@ -12,20 +12,26 @@
-    // TODO: Set up the Environment File ID from your Launch property for the preferred environment
-    private let LAUNCH_ENVIRONMENT_FILE_ID = ""
+    // TODO: Set up the preferred Environment File ID from your mobile property configured in Data Collection UI
+    private let ENVIRONMENT_FILE_ID = ""
```
private let ENVIRONMENT_FILE_ID = "yourAppId"

To match Android.

emdobrin replied:
thanks Calise, we can update that in a future PR

emdobrin committed:
Merge branch 'dev' into feature/adIdConsent
